zerver/lib/streams.py

```diff
@@ -511,17 +520,28 @@ def list_to_streams(streams_raw: Iterable[StreamDict],
     """
     # Validate all streams, getting extant ones, then get-or-creating the rest.
 
-    stream_set = {stream_dict["name"] for stream_dict in streams_raw}
+    stream_set = {stream_dict["stream"] for stream_dict in streams_raw}
+    is_stream_type_name = isinstance(next(iter(stream_set), None), str)
+
+    if is_stream_type_name:
+        for stream_name in stream_set:
+            # Stream names should already have been stripped by the
+            # caller, but it makes sense to verify anyway.
+            assert isinstance(stream_name, str) and stream_name == stream_name.strip()
+            check_stream_name(stream_name)
 
-    for stream_name in stream_set:
-        # Stream names should already have been stripped by the
-        # caller, but it makes sense to verify anyway.
-        assert stream_name == stream_name.strip()
-        check_stream_name(stream_name)
+    else:
+        for stream_id in stream_set:
+            # Stream id should already be parsed in int by the caller,
+            # but it makes sense to verify anyway.
+            assert isinstance(stream_id, int)
 
     existing_streams: List[Stream] = []
     missing_stream_dicts: List[StreamDict] = []
-    existing_stream_map = bulk_get_streams_by_names(user_profile.realm, stream_set)
+    existing_stream_map = bulk_get_streams_by_names(user_profile.realm, cast(Set[str], stream_set)) if is_stream_type_name else bulk_get_streams_by_ids(user_profile.realm, cast(Set[int], stream_set))
+    # Casting it to Dict[Union[int, str], Any] beacuse inline if/else
+    # (foo = x if y else z) returns object.
+    existing_stream_map = cast(Dict[Union[int, str], Any], existing_stream_map)
```

Comment:
Would it be better to just use the normal `if else` syntax here? (Since the line length is also quiet long here)